src/Login/index.test.js

```diff
@@ -1,15 +1,75 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { fireEvent, render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import { BrowserRouter as Router } from 'react-router-dom';
+import { MemoryRouter as Router, Switch, Route } from 'react-router-dom';
+import { MockedProvider } from '@apollo/client/testing';
+import AuthProvider from '../AuthProvider';
 import Login from './index';
+import QUERY_USERS from '../graphql/queries';
+import { HOME_PATH, LOGIN_PATH } from '../paths';
 
-it('renders the login page', () => {
-  render(<Router><Login /></Router>);
+const mockedUsersResponse = {
+  request: { query: QUERY_USERS },
+  result: {
+    data: {
+      users: [
+        {
+          username: 'User',
+          password: 'password',
+          id: 1,
+        },
+      ],
+    },
+  },
+};
+
+const mocks = [
+  mockedUsersResponse,
+  mockedUsersResponse,
+  mockedUsersResponse,
+];
+
+it('renders the login page', async () => {
+  render(
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <AuthProvider>
+        <Router initialEntries={[LOGIN_PATH]}>
+          <Switch>
+            <Route
+              exact
+              path={HOME_PATH}
+              component={() => <div>Home</div>}
+            />
+            <Route path={LOGIN_PATH} component={Login} />
+          </Switch>
+        </Router>
+      </AuthProvider>
+    </MockedProvider>,
+  );
   expect(screen.getByRole('form')).toBeInTheDocument();
-  expect(screen.getByText('Register')).toBeInTheDocument();
-  userEvent.type(screen.getByLabelText('Username'), 'User');
-  expect(screen.getByDisplayValue('User')).toBeInTheDocument();
-  userEvent.type(screen.getByLabelText('Password'), 'password');
+  expect(screen.getByText('Register Now')).toBeInTheDocument();
+  const [loginButton] = screen.getAllByRole('button');
+  expect(loginButton).toBeDisabled();
+
+  const usernameField = screen.getByLabelText('Username');
+  userEvent.type(usernameField, ' {backspace}');
+  expect(await screen.findByText('Username required.')).toBeInTheDocument();
+  userEvent.type(usernameField, 'User_');
+  expect(screen.getByDisplayValue('User_')).toBeInTheDocument();
+
+  const passwordField = screen.getByLabelText('Password');
+  userEvent.type(passwordField, ' {backspace}');
+  expect(await screen.findByText('Password required.')).toBeInTheDocument();
+  userEvent.type(passwordField, 'password');
   expect(screen.getByDisplayValue('password')).toBeInTheDocument();
+
+  expect(loginButton).toBeEnabled();
+
+  fireEvent.click(loginButton);
```

Comment:
You can use `userEvent` here instead of `fireEvent`, and it's usually a best practice to always use `userEvent` whenever is possible since it's what's close to mimicking user behavior, check this article that explains the reasons: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event